config.go

```diff
@@ -216,21 +216,42 @@ func (c *SysConfig) WithEnv(key, value string) *SysConfig {
 
 // WithFS assigns the file system to use for any paths beginning at "/". Defaults to not found.
 //
+// Ex. This sets a read-only, embedded file-system to serve files under the root ("/") and working (".") directories:
+//
+//	//go:embed testdata/index.html
+//	var testdataIndex embed.FS
+//
+//	rooted, err := fs.Sub(testdataIndex, "testdata")
+//	require.NoError(t, err)
+//
+//	// "index.html" is accessible as both "/index.html" and "./index.html" because we didn't use WithWorkDirFS.
+//	sysConfig := wazero.NewSysConfig().WithFS(rooted)
+//
 // Note: This sets WithWorkDirFS to the same file-system unless already set.
-func (c *SysConfig) WithFS(fs wasi.FS) *SysConfig {
+func (c *SysConfig) WithFS(fs fs.FS) *SysConfig {
 	c.setFS("/", fs)
 	return c
 }
 
-// WithWorkDirFS indicates the file system to use for any paths beginning at ".". Defaults to the same as WithFS.
-func (c *SysConfig) WithWorkDirFS(fs wasi.FS) *SysConfig {
+// WithWorkDirFS indicates the file system to use for any paths beginning at "./". Defaults to the same as WithFS.
+//
+// Ex. This sets a read-only, embedded file-system as the root ("/"), and a mutable one as the working directory ("."):
+//
+//	//go:embed appA
+//	var rootFS embed.FS
+//
+//	// Files relative to this source under appA is available under "/" and files relative to "/work/appA" under ".".
+//	sysConfig := wazero.NewSysConfig().WithFS(rootFS).WithWorkDirFS(os.DirFS("/work/appA"))
+//
+// Note: os.DirFS documentation includes important notes about isolation, which also applies to fs.Sub. As of Go 1.18,
```

Comment:
took some reading to discover there are no plans for an actual chrooted FS. Until then, folks with high isolation needs will have to write their own fs.FS impl (or find some library that isn't abandoned that tries to do this!)